### PR TITLE
[MXNET-121] Docs page for ONNX module.

### PR DESCRIPTION
**Pull request by anirudhacharya**
## Description ##
Documentation page for ONNX module.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Documentation page for onnx module in contrib.

## Comments ##
- @reminisce @nswamy @Roshrini @spidyDev

### CODE REVIEW

File: docs/api/python/contrib/onnx.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
+# ONNX API
```
Comment (nswamy) [inline docs/api/python/contrib/onnx.md]: ONNX-MXNet Converter API ? 
Please check the rendered output and fix the markdown

I think this can be created as a notebook, which users can use as a tutorial

File: docs/api/python/contrib/onnx.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+The `mxnet.contrib.onnx` package refers to the APIs and interfaces that implements ONNX model format support for Apache MXNet.
+
+With ONNX format support for MXNet, developers can build and train models with PyTorch, CNTK, or Caffe2, and import these models into MXNet to run them for inference using MXNet’s highly optimized engine.
+
+```eval_rst
+.. warning:: This package contains experimental APIs and may change in the near future.
+```
+
+```eval_rst
+.. note:: **Install ONNX** which needs protobuf compiler to be installed separately. Please **follow the instructions to install ONNX** [here](https://github.com/onnx/onnx).
```
Comment (nswamy) [inline docs/api/python/contrib/onnx.md]: mark down render check?

File: docs/api/python/contrib/onnx.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+```
+
+This document describes the ONNX APIs in mxnet.
```
Comment (nswamy) [inline docs/api/python/contrib/onnx.md]: ONNX-MXNet API

File: docs/api/python/contrib/onnx.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    mxnet.contrib.onnx.import_model
+```
+
+## Import ONNX model into Mxnet
```
Comment (nswamy) [inline docs/api/python/contrib/onnx.md]: Using ONNX Model in MXNet

File: docs/api/python/contrib/onnx.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+```python
+pip install Pillow
+```
+
+We will now try to import a [Super_Resolution model](http://pytorch.org/tutorials/advanced/super_resolution_with_caffe2.html), trained with PyTorch, and run inference in MXNet. PyTorch provides a way to export models in ONNX protobuf format. Using this functionality, we have exported the model into ONNX format.
+
+You can download the converted ONNX model from
+[here](https://s3.amazonaws.com/onnx-mxnet/examples/super_resolution.onnx).
+
+A pre-trained model in MXNet contains two elements: a symbolic graph, containing the model's network definition, and a binary file containing the model weights. You can import the ONNX model and get the symbol and parameters objects using "import_model" API as shown below:
```
Comment (nswamy) [inline docs/api/python/contrib/onnx.md]: bulleted points
* Symbolic graph(with the extension .json) - containing the model's network definition
* Binary file(with the extension .params) - containing model weights.
..

File: docs/api/python/contrib/onnx.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+img = Image.open('super_res_input.jpg').resize((224, 224))
+img_ycbcr = img.convert("YCbCr")
+img_y, img_cb, img_cr = img_ycbcr.split()
+test_image = np.array(img_y)[np.newaxis, np.newaxis, :, :]
+```
+- We'll be using MXNet's Module API to create the module, bind it and assign the loaded weights.
+
+```
+# By default, 'input_0' is an input of the imported model.
+mod = mx.mod.Module(symbol=sym, data_names=['input_0'], context=mx.cpu(), label_names=None)
```
Comment (nswamy) [inline docs/api/python/contrib/onnx.md]: link to Module API/tutorial page, similarly others where useful

**Comment by ThomasDelteil** [comment]
I am writing a tutorial on ONNX with MXNet, it would be nice if the docs show an example on how to load an ONNX model in Gluon using a symbolic block.

**Comment by anirudhacharya** [comment]
@ThomasDelteil Will include it. As discussed on github, we will also eventually have an API for gluon import and that API will have its own documentation and example illustration.

**Comment by anirudhacharya** [comment]
@aaronmarkham

File: docs/api/python/contrib/onnx.md (hunk 3 above)
Comment (spidyDev) [reply docs/api/python/contrib/onnx.md]: This seems a bit confusing as we are describing ONNX API's . May be something like 
"This document describes APIs to support ONNX in MXNet"

**Comment by anirudhacharya** [comment]
This is how the page renders - http://34.227.67.197/api/python/contrib/onnx.html
Please review @aaronmarkham @sandeep-krishnamurthy @nswamy

File: docs/api/python/contrib/onnx.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
+# ONNX-MXNet API
+
+## Overview
+
+[ONNX](https://onnx.ai/) is an open format to represent deep learning models. With ONNX as an intermediate representation, it is easier to move models between state-of-the-art tools and frameworks for training and inference..
```
Comment (aaronmarkham) [inline docs/api/python/contrib/onnx.md]: Remove extra period

File: docs/api/python/contrib/onnx.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+The `mxnet.contrib.onnx` package refers to the APIs and interfaces that implements ONNX model format support for Apache MXNet.
```
Comment (aaronmarkham) [inline docs/api/python/contrib/onnx.md]: implement

File: docs/api/python/contrib/onnx.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+With ONNX format support for MXNet, developers can build and train models with PyTorch, CNTK, or Caffe2, and import these models into MXNet to run them for inference and training using MXNet’s highly optimized engine.
```
Comment (aaronmarkham) [inline docs/api/python/contrib/onnx.md]: Simplify.
...train models with a variety of deep learning frameworks...
Then list the currently supported frameworks with bullets. (easier to maintain too, when we add more later)

File: docs/api/python/contrib/onnx.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+```eval_rst
+.. warning:: This package contains experimental APIs and may change in the near future.
+```
+
+```eval_rst
+.. note:: To use this module developers need to **install ONNX**, which needs protobuf compiler, to be installed separately. Please follow the instructions to install ONNX - https://github.com/onnx/onnx
```
Comment (aaronmarkham) [inline docs/api/python/contrib/onnx.md]: Make this installation instructions.

File: docs/tutorials/onnx/super_resolution.md [hunk 11]
```diff
@@ -0,0 +1,113 @@
+# Importing an ONNX model into MXNet
+
+In this tutorial we will:
+
+- learn how to load a pre-trained ONNX model file into MXNet.
+- run inference in MXNet.
+
+## Pre-requisite
```
Comment (aaronmarkham) [inline docs/tutorials/onnx/super_resolution.md]: plural, no dash

File: docs/tutorials/onnx/super_resolution.md [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+The code demonstration assumes that the following python packages are installed:
```
Comment (aaronmarkham) [inline docs/tutorials/onnx/super_resolution.md]: This example assumes...

File: docs/tutorials/onnx/super_resolution.md [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+## Loading the model into MXNet
+
+To completely describe a pre-trained model in MXNet, we need two elements: a symbolic graph, containing the model's network definition, and a binary file containing the model weights. You can import the ONNX model and get the symbol and parameters objects using ``import_model`` API. The paameter object is split into argument parameters and auxilliary parameters.
+
+
+```python
+sym, arg, aux = onnx_mxnet.import_model(onnx_model_file)
+```
+
+We can now visualize the imported model( graphviz needs to be installed)
```
Comment (aaronmarkham) [inline docs/tutorials/onnx/super_resolution.md]: space before ( not after

File: docs/tutorials/onnx/super_resolution.md [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+We will use MXNet's Module API to run the inference. For this we will need to create the module, bind it to the input data and assign the loaded weights from the two parameter objects - argument parameters and auxilliary parameters.
+
+
+```python
+mod = mx.mod.Module(symbol=sym, data_names=['input_0'], context=mx.cpu(), label_names=None)
+mod.bind(for_training=False, data_shapes=[('input_0',test_image.shape)], label_shapes=None)
+mod.set_params(arg_params=arg, aux_params=aux, allow_missing=True, allow_extra=True)
+```
+
+Module API's forward method requires Batch of data as input. We will prepare the data in that format and feed it to the forward method.
```
Comment (aaronmarkham) [inline docs/tutorials/onnx/super_resolution.md]: lower case b

File: docs/api/python/contrib/onnx.md (hunk 9 above)
Comment (anirudhacharya) [reply docs/api/python/contrib/onnx.md]: How about this -
... train models with a [variety of deep learning frameworks](http://onnx.ai/supported-tools)....
I have linked it to the supported tools page on the ONNX website. That way our page never gets outdated.

File: docs/api/python/contrib/onnx.md [hunk 15]
```diff
@@ -0,0 +1,49 @@
+# ONNX-MXNet API
+
+## Overview
+
+[ONNX](https://onnx.ai/) is an open format to represent deep learning models. With ONNX as an intermediate representation, it is easier to move models between state-of-the-art tools and frameworks for training and inference.
+
+The `mxnet.contrib.onnx` package refers to the APIs and interfaces that implement ONNX model format support for Apache MXNet.
+
+With ONNX format support for MXNet, developers can build and train models with a [variety of deep learning frameworks](http://onnx.ai/supported-tools), and import these models into MXNet to run them for inference and training using MXNet’s highly optimized engine.
```
Comment (sandeep-krishnamurthy) [inline docs/api/python/contrib/onnx.md]: also for re training and transfer learning usecases right?

Comment (anirudhacharya) [reply docs/api/python/contrib/onnx.md]: yes
